tests/unittests/general/webfront_test.py

```diff
@@ -19,16 +19,16 @@ class TestLdapAuthenticate(object):
     def test_authenticate_should_return_account_when_ldap_says_yes(self):
         with patch("nav.web.ldapauth.available", new=True):
             with patch("nav.web.ldapauth.authenticate", return_value=True):
-                assert auth.authenticate('knight', 'shrubbery') == LDAP_ACCOUNT
+                assert auth.authenticate_ldap('knight', 'shrubbery') == LDAP_ACCOUNT
 
     def test_authenticate_should_return_false_when_ldap_says_no(self):
         with patch("nav.web.ldapauth.available", new=True):
             with patch("nav.web.ldapauth.authenticate", return_value=False):
-                assert not auth.authenticate('knight', 'shrubbery')
+                assert auth.authenticate_ldap('knight', 'shrubbery') is None
 
     def test_authenticate_should_fallback_when_ldap_is_disabled(self):
         with patch("nav.web.ldapauth.available", new=False):
-            assert auth.authenticate('knight', 'shrubbery') == LDAP_ACCOUNT
+            assert auth.authenticate_ldap('knight', 'shrubbery') is None
```

Comment:
This test certainly no longer tests what it says. If you change the behaviour of the code so that a test is completely irrelevant, it should be removed rather than changed to mindlessly accept the new behaviour

Reply:
In the body of the commit I wrote something about this. My mock/patch-skills aren't up to only patching the `Account.object.get` that may happen in `authenticate_ldap` if patching `authenticate`. I still haven't figured out how to test that an account is created either, but at least now it should be easier.

Reply:
This doesn't have anything to do with your mock/patch skills. The test says that the authenticate function should fall back when ldap is disabled, and the test does not even call the authenticate function, and most certainly does not check that it falls back. This entire test becomes a complete lie and serves no purpose but to mislead developers

Reply:
Added new tests that specifically end explicitly tests what auth.authenticate is to do.